plugins/org.jkiss.dbeaver.ext.altibase/src/org/jkiss/dbeaver/ext/altibase/model/plan/AltibasePlanBuilder.java

```diff
@@ -50,87 +58,218 @@ public static List<AltibasePlanNode> build(AltibaseDataSource dataSource, String
+            } else {
+                parentNode = node.getParentNodeAtDepth(depth);
             }
+        } else { 
+            throw new IllegalStateException("Plan parsing error [depth: " + depth + "]: " + nodeStr );
+        }
 
-            prevNode = node;
+        newNode = new AltibasePlanNode(dataSource, idIdx, depth, nodeStr.toString(), parentNode);
+        if (depth == 0) {
+            rootNodes.add(newNode);
         }
 
-        return rootNodes;
+        return newNode;
     }
 
     /**
      * Test code
      */
     public static void main(String[] args) {
-        String plan = "-----------------------------------------------------------" + "\n"
-                + "PROJECT ( COLUMN_COUNT: 2, TUPLE_SIZE: 34, COST: 151146.46 )" + "\n"
-                + " JOIN ( METHOD: NL, COST: 148444.31 )" + "\n"
-                + "  SCAN ( TABLE: CUSTOMERS, FULL SCAN, ACCESS: ??, COST: 116.76 )" + "\n"
-                + "  VIEW ( ORDERS_T, ACCESS: ??, COST: 14.49 )" + "\n"
-                + "   PROJECT ( COLUMN_COUNT: 6, TUPLE_SIZE: 48, COST: 2.81 )" + "\n"
-                + "    VIEW ( ACCESS: ??, COST: 2.02 )" + "\n"
-                + "     BAG-UNION" + "\n"
-                + "      PROJECT ( COLUMN_COUNT: 6, TUPLE_SIZE: 48, COST: 0.67 )" + "\n"
-                + "       SCAN ( TABLE: ORDERS ORDERS_01, INDEX: SYS.ODR_IDX2, RANGE SCAN, ACCESS: ??, COST: 0.41 )" + "\n"
-                + "        [ VARIABLE KEY ]" + "\n"
-                + "        OR" + "\n"
-                + "         AND" + "\n"
-                + "          CUSTOMERS.CNO = ORDERS_01.CNO" + "\n"
-                + "        [ FILTER ]" + "\n"
-                + "        ORDERS_01.QTY >= 10000" + "\n"
-                + "      PROJECT ( COLUMN_COUNT: 6, TUPLE_SIZE: 48, COST: 0.67 )" + "\n"
-                + "       SCAN ( TABLE: ORDERS ORDERS_02, INDEX: SYS.ODR_IDX2, RANGE SCAN, ACCESS: ??, COST: 0.41 )" + "\n"
-                + "        [ VARIABLE KEY ]" + "\n"
-                + "        OR" + "\n"
-                + "         AND" + "\n"
-                + "          CUSTOMERS.CNO = ORDERS_02.CNO" + "\n"
-                + "        [ FILTER ]" + "\n"
-                + "        ORDERS_02.QTY >= 10000" + "\n"
-                + "      PROJECT ( COLUMN_COUNT: 6, TUPLE_SIZE: 48, COST: 0.67 )" + "\n"
-                + "       SCAN ( TABLE: ORDERS ORDERS_03, INDEX: ODR_IDX2, RANGE SCAN, ACCESS: ??, COST: 0.41 ) " + "\n"
-                + "        [ VARIABLE KEY ]" + "\n"
-                + "        OR" + "\n"
-                + "         AND" + "\n"
-                + "          CUSTOMERS.CNO = ORDERS_03.CNO" + "\n"
-                + "        [ FILTER ]" + "\n"
-                + "        ORDERS_03.QTY >= 10000" + "\n"
-                + "-----------------------------------------------------------"
-                + "* SIMPLE QUERY PLAN";
+        String plan =
```

Comment:
Tests are cool, but can we convert them to actual tests?

Tests are located [here](https://github.com/dbeaver/dbeaver/tree/devel/test). You can create a new bundle and move tests there.

Reply:
Hello,  @ShadelessFox

Thanks for your comment. I will implement the test according to your advice.

Have a great day :)

Reply:
Hi @ShadelessFox,

Could you let me know what kind of archetype should be used to create a test Maven module?

tests/pom.xml -> New Maven Module Project -> org.jkiss.dbeaver.ext.altibase.test -> **_Select an Archetype_** ?

Reply:
The archetype is not required.

I don't use Eclipse for development, so I'm unsure why this error occurs. Have you tried building DBeaver from the command line?

Reply:
Thanks for your reply :)

Building DBeaver with build.sh at command line looks working correctly as well as my new test. The problem may comes from Eclipse or maven cache. I'll try to find the root cause.

![image](https://github.com/dbeaver/dbeaver/assets/32251385/260265ec-cbd0-4ea4-a2f2-0585303b909f)

![image](https://github.com/dbeaver/dbeaver/assets/32251385/0439d9e0-6344-4e49-8125-b5a413adcac7)

Reply:
I tried to figure out the the error at pom.xml, but I couldn't. (https://stackoverflow.com/questions/65677156/could-not-transfer-artifact-from-to-central-intellij) But, what I know is that it comes from not my code but m2 repository cache.

When I tried execute 'mvn clean verify' at dbeaver/test or build.sh at dbeaver at command line, everything is fine. So, I committed my new test bundle. 

If any problem found, please let me know.

Thanks,

=================================
Error message
=================================
org.slf4j:slf4j-api:jar:1.7.36 failed to transfer from https://repo.maven.apache.org/maven2 during a previous attempt. 
This failure was cached in the local repository and resolution is not reattempted until the update interval of central has elapsed or updates are forced. 
Original error: Could not transfer artifact org.slf4j:slf4j-api:jar:1.7.36 from/to central (https://repo.maven.apache.org/maven2): The operation was cancelled.

Reply:
Thank you

Reply:
> Thank you

I should say thank you for your help :)